### PR TITLE
Add status checker

### LINKED ISSUE #10: Idea: Port forwarding checking command
Add a command like `!pfcheck ip:port` or/and `!pf ip:port` to easily check port forwarding on a specific server. 
Maybe also automatically detect server IPs and Adresses with "not working" context and check port forwarding for them. If it is an alphaberic adress detect if its anything than an A record and add a note about it.

### PR DESCRIPTION
**Pull request by Heath123**
Closes #10

### CODE REVIEW

File: bot_modules/server_status_checker/index.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        }
+      }
+    }
+  })
+}
+
+/**
+ * Pings the Java server and sends an embed to the channel specified
+ *
+ * @param {Discord.Channel} channel TODO: TODO TODO TODO
```
Comment (rtm516) [inline bot_modules/server_status_checker/index.js:40]: This documentation line needs filling out

File: bot_modules/server_status_checker/bedrockProtocolVersion.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+let xPathSelect
+
+async function init () {
+  // https://stackoverflow.com/questions/25753368/performant-parsing-of-pages-with-node-js-and-xpath
+
+  const html = (await Utils.getContents('https://minecraft.gamepedia.com/Template:Protocol_version/Table/doc')).data
+  const document = parse5.parse(html.toString())
+  const xhtml = xmlser.serializeToString(document)
+  wikiDoc = new Dom().parseFromString(xhtml)
+  xPathSelect = xpath.useNamespaces({ x: 'http://www.w3.org/1999/xhtml' })
```
Comment (rtm516) [inline bot_modules/server_status_checker/bedrockProtocolVersion.js:18]: Could this be replaced with cheerio? (To save the 4 extra dependencies)

File: bot_modules/server_status_checker/bedrockProtocolVersion.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+init()
+
+/**
+ * Converts a protocol version number to a name
+ *
+ * @param {Number|String} protocolVersion The protocol number
+ */
+exports.versionFromProtocol = async function (protocolVersion) {
+  const nodes = xPathSelect(`//*[@id="mw-content-text"]/x:div/x:table[5]//x:td[text()='${protocolVersion}']`, wikiDoc)
+  return nodes[0].parentNode.firstChild.textContent.replace('Bedrock Edition ', '')
```
Comment (rtm516) [inline bot_modules/server_status_checker/bedrockProtocolVersion.js:30]: Same as above

File: utils.js [hunk 4]
```diff
@@ -36,3 +49,63 @@ exports.postContents = async (url, contents) => {
...
+
+/**
+ * Ping a Minecraft Bedrock Edition server and return the response
+ *
+ * @param {String} address The address of the Minecraft server to ping
+ * @param {Number} port The port of the Mincraft server to ping
+ * @param {Number} timeout The time to wait in milliseconds
+ */
+exports.pingBedrockServer = function (address, port, timeout) {
+  console.log(address, port)
```
Comment (rtm516) [inline utils.js:79]: Left in debug log?

**Review by rtm516 — CHANGES_REQUESTED** [review]

File: bot_modules/server_status_checker/bedrockProtocolVersion.js (hunk 2 above)
Comment (Heath123) [reply bot_modules/server_status_checker/bedrockProtocolVersion.js:18]: I'll see if can

File: utils.js (hunk 4 above)
Comment (Heath123) [reply utils.js]: Yep, fixed now

**Pull request closed without merging**